boa/src/environment/global_environment_record.rs

```diff
@@ -53,221 +53,465 @@ impl GlobalEnvironmentRecord {
+
+        // 3. Let hasProperty be ? HasOwnProperty(globalObject, N).
+        let has_property = global_object.has_own_property(name, context)?;
+        // 4. Let extensible be ? IsExtensible(globalObject).
+        let extensible = global_object.is_extensible(context)?;
+
+        // 5. If hasProperty is false and extensible is true, then
         if !has_property && extensible {
-            obj_rec.create_mutable_binding(name.clone(), deletion, false, context)?;
-            obj_rec.initialize_binding(&name, JsValue::undefined(), context)?;
+            // a. Perform ? ObjRec.CreateMutableBinding(N, D).
+            self.object_record
+                .create_mutable_binding(name, deletion, false, context)?;
+            // b. Perform ? ObjRec.InitializeBinding(N, undefined).
+            self.object_record
+                .initialize_binding(name, JsValue::undefined(), context)?;
         }
 
+        // 6. Let varDeclaredNames be envRec.[[VarNames]].
         let mut var_declared_names = self.var_names.borrow_mut();
-        if !var_declared_names.contains(name.as_str()) {
-            var_declared_names.insert(name.into_boxed_str());
+        // 7. If varDeclaredNames does not contain N, then
+        if !var_declared_names.contains(name) {
+            // a. Append N to varDeclaredNames.
+            var_declared_names.insert(name.into());
         }
+
+        // 8. Return NormalCompletion(empty).
         Ok(())
     }
 
-    pub fn create_global_function_binding(&mut self, name: &str, value: JsValue, deletion: bool) {
-        let global_object = &mut self.object_record.bindings;
-        let existing_prop = global_object.get_property(name);
-        // TODO: change to desc.is_undefined()
-        let desc = match existing_prop {
-            Some(desc) if desc.expect_configurable() => PropertyDescriptor::builder().value(value),
-            Some(_) => PropertyDescriptor::builder()
-                .value(value)
+    /// `9.1.1.4.18 CreateGlobalFunctionBinding ( N, V, D )`
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-createglobalfunctionbinding
+    pub fn create_global_function_binding(
+        &mut self,
+        name: &str,
+        value: JsValue,
+        deletion: bool,
+        context: &mut Context,
+    ) -> JsResult<()> {
+        // 1. Let ObjRec be envRec.[[ObjectRecord]].
+        // 2. Let globalObject be ObjRec.[[BindingObject]].
+        let global_object = &self.object_record.bindings;
+
+        // 3. Let existingProp be ? globalObject.[[GetOwnProperty]](N).
+        let existing_prop = global_object.__get_own_property__(&name.into(), context)?;
+
+        // 4. If existingProp is undefined or existingProp.[[Configurable]] is true, then
+        let desc = if existing_prop.is_none()
+            || existing_prop
+                .map(|f| f.configurable().unwrap_or(false))
+                .unwrap_or(false)
```

Comment:
```suggestion
        let desc = if existing_prop
                .map(|f| f.expect_configurable())
                .unwrap_or(true)
```
You can unwrap with true and is the same as `existing_prop.is_none()`